### PR TITLE
Fix layout for users without username but with common friends

### PR DESCRIPTION
**Pull request by daehn**
# What's in this PR?

* Fix padding in `UserDetailView`.
* Use `UserDetailView` in `IncomingConnectionView` instead of repeating the logic there.
* Use the first label for the correlation text if the user does not have an username.
* Add accessibility identifier to `SearchResultsCell`.

### CODE REVIEW

File: Wire-iOS/Sources/UserInterface/ConnectRequests/IncomingConnectionView.swift [hunk 1]
```diff
@@ -30,21 +30,15 @@ public final class IncomingConnectionView: UIView {
...
-    private let labelContainer = UIView()
-
+    private let userDetailView = UserNameDetailView()
     private let userImageView = UserImageView()
     private let incomingConnectionFooter = UIView()
     private let acceptButton = Button(style: .full)
     private let ignoreButton = Button(style: .empty)
 
-    public var user: User {
+    public var user: ZMUser {
```
Comment (daehn) [inline Wire-iOS/Sources/UserInterface/ConnectRequests/IncomingConnectionView.swift]: Removing the `ZMBareUser & ZMBareUserConnection & ZMSearchableUser` was related to a change that will come with the PR using the new address book name API.

Comment (daehn) [inline Wire-iOS/Sources/UserInterface/ConnectRequests/IncomingConnectionView.swift]: Ah I see, still works in the tests passing in a `MockUser`.

File: Wire-iOS/Sources/UserInterface/ConnectRequests/UserConnectionView.swift [hunk 2]
```diff
@@ -66,48 +64,53 @@ public final class UserConnectionView: UIView, Copyable {
...
     private func setup() {
-        [handleLabel, correlationLabel].forEach {
+        [firstLabel, secondLabel].forEach {
             $0.numberOfLines = 0
             $0.textAlignment = .center
         }
 
-        handleLabel.accessibilityIdentifier = "handle"
-        correlationLabel.accessibilityIdentifier = "correlation"
+        firstLabel.accessibilityIdentifier = "handle"
```
Comment (mikeger) [inline Wire-iOS/Sources/UserInterface/ConnectRequests/UserConnectionView.swift]: The accessibility label is not correct any more.

Comment (daehn) [inline Wire-iOS/Sources/UserInterface/ConnectRequests/UserConnectionView.swift]: Good catch, thanks!

**Comment by daehn** [comment]
@mikeger please re-review!

**Pull request merged**